a test is flaky in pipedv1/controller/planner_test.go

**What happened**:

- https://github.com/pipe-cd/pipecd/blob/3d4bd4f2a330168b1ee246ce2b2aeb710174c3d1/pkg/app/pipedv1/controller/planner_test.go#L455-L456 seems a flaky test.

**What you expected to happen**:

- not flaky

**How to reproduce it**:

Retry the test again and again.

**Environment**:
- `piped` version: N/A
- `control-plane` version: N/A
- Others:     reproduced in https://github.com/pipe-cd/pipecd/commit/3d4bd4f2a330168b1ee246ce2b2aeb710174c3d1

**log**:

```
$ go test ./pkg/app/pipedv1/controller/ -count=1 
ok      github.com/pipe-cd/pipecd/pkg/app/pipedv1/controller     0.250s

$ go test ./pkg/app/pipedv1/controller/ -count=1
ok      github.com/pipe-cd/pipecd/pkg/app/pipedv1/controller     0.246s

$ go test ./pkg/app/pipedv1/controller/ -count=1
ok      github.com/pipe-cd/pipecd/pkg/app/pipedv1/controller     0.251s

$ go test ./pkg/app/pipedv1/controller/ -count=1
--- FAIL: TestBuildPipelineSyncStages (0.00s)
    --- FAIL: TestBuildPipelineSyncStages/multi_plugins_multi_rollback (0.00s)
        planner_test.go:583: 
                Error Trace:    /Users/xxx/pipecd/upstream/pipecd/pkg/app/pipedv1/controller/planner_test.go:583
                Error:          Not equal: 
                                expected: []*model.PipelineStage{(*model.PipelineStage)(0x14000121200), (*model.PipelineStage)(0x140001212c0), (*model.PipelineStage)(0x14000121380), (*model.PipelineStage)(0x14000121440), (*model.PipelineStage)(0x14000121500), (*model.PipelineStage)(0x140001215c0)}
                                actual  : []*model.PipelineStage{(*model.PipelineStage)(0x140001206c0), (*model.PipelineStage)(0x14000120780), (*model.PipelineStage)(0x140001209c0), (*model.PipelineStage)(0x14000120840), (*model.PipelineStage)(0x14000120a80), (*model.PipelineStage)(0x14000120900)}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -127,4 +127,4 @@
                                   unknownFields: ([]uint8) <nil>,
                                -  Id: (string) (len=17) "plugin-1-rollback",
                                -  Name: (string) (len=17) "plugin-1-rollback",
                                +  Id: (string) (len=17) "plugin-2-rollback",
                                +  Name: (string) (len=17) "plugin-2-rollback",
                                   Desc: (string) "",
                                @@ -154,4 +154,4 @@
                                   unknownFields: ([]uint8) <nil>,
                                -  Id: (string) (len=17) "plugin-2-rollback",
                                -  Name: (string) (len=17) "plugin-2-rollback",
                                +  Id: (string) (len=17) "plugin-1-rollback",
                                +  Name: (string) (len=17) "plugin-1-rollback",
                                   Desc: (string) "",
                Test:           TestBuildPipelineSyncStages/multi_plugins_multi_rollback
FAIL
FAIL    github.com/pipe-cd/pipecd/pkg/app/pipedv1/controller     0.248s
FAIL
```

--- a/pkg/app/pipedv1/controller/planner.go
+++ b/pkg/app/pipedv1/controller/planner.go
@@ -478,6 +478,7 @@ func (p *planner) buildPipelineSyncStages(ctx context.Context, cfg *config.Gener
 
 	// Sort stages by index.
 	sort.Sort(model.PipelineStages(stages))
+	sort.Sort(model.PipelineStages(rollbackStages))
 
 	// Build requires for each stage.
 	preStageID := ""

--- a/pkg/app/pipedv1/controller/planner_test.go
+++ b/pkg/app/pipedv1/controller/planner_test.go
@@ -148,12 +148,14 @@ func TestBuildQuickSyncStages(t *testing.T) {
 					quickStages: []*model.PipelineStage{
 						{
 							Id:      "plugin-1-stage-1",
+							Index:   1,
 							Visible: true,
 						},
 					},
 					rollbackStages: []*model.PipelineStage{
 						{
 							Id:      "plugin-1-rollback",
+							Index:   1,
 							Visible: false,
 						},
 					},
@@ -162,12 +164,14 @@ func TestBuildQuickSyncStages(t *testing.T) {
 					quickStages: []*model.PipelineStage{
 						{
 							Id:      "plugin-2-stage-1",
+							Index:   2,
 							Visible: true,
 						},
 					},
 					rollbackStages: []*model.PipelineStage{
 						{
 							Id:      "plugin-2-rollback",
+							Index:   2,
 							Visible: false,
 						},
 					},
@@ -182,18 +186,22 @@ func TestBuildQuickSyncStages(t *testing.T) {
 			expectedStages: []*model.PipelineStage{
 				{
 					Id:      "plugin-1-stage-1",
+					Index:   1,
 					Visible: true,
 				},
 				{
 					Id:      "plugin-2-stage-1",
+					Index:   2,
 					Visible: true,
 				},
 				{
 					Id:      "plugin-1-rollback",
+					Index:   1,
 					Visible: false,
 				},
 				{
 					Id:      "plugin-2-rollback",
+					Index:   2,
 					Visible: false,
 				},
 			},
@@ -458,16 +466,19 @@ func TestBuildPipelineSyncStages(t *testing.T) {
 					pipelineStages: []*model.PipelineStage{
 						{
 							Id:      "plugin-1-stage-1",
+							Index:   0,
 							Name:    "plugin-1-stage-1",
 							Visible: true,
 						},
 						{
 							Id:      "plugin-1-stage-2",
+							Index:   1,
 							Name:    "plugin-1-stage-2",
 							Visible: true,
 						},
 						{
 							Id:      "plugin-1-stage-3",
+							Index:   2,
 							Name:    "plugin-1-stage-3",
 							Visible: true,
 						},
@@ -475,6 +486,7 @@ func TestBuildPipelineSyncStages(t *testing.T) {
 					rollbackStages: []*model.PipelineStage{
 						{
 							Id:      "plugin-1-rollback",
+							Index:   0,
 							Name:    "plugin-1-rollback",
 							Visible: false,
 						},
@@ -484,6 +496,7 @@ func TestBuildPipelineSyncStages(t *testing.T) {
 					pipelineStages: []*model.PipelineStage{
 						{
 							Id:      "plugin-2-stage-1",
+							Index:   3,
 							Name:    "plugin-2-stage-1",
 							Visible: true,
 						},
@@ -491,6 +504,7 @@ func TestBuildPipelineSyncStages(t *testing.T) {
 					rollbackStages: []*model.PipelineStage{
 						{
 							Id:      "plugin-2-rollback",
+							Index:   3,
 							Name:    "plugin-2-rollback",
 							Visible: false,
 						},
@@ -553,11 +567,13 @@ func TestBuildPipelineSyncStages(t *testing.T) {
 				},
 				{
 					Id:      "plugin-1-rollback",
+					Index:   0,
 					Name:    "plugin-1-rollback",
 					Visible: false,
 				},
 				{
 					Id:      "plugin-2-rollback",
+					Index:   3,
 					Name:    "plugin-2-rollback",
 					Visible: false,
 				},

